policybot/TEST_FLAKES_BOT.md

```diff
@@ -0,0 +1,31 @@
+# Test Flakes Bot design
+
+## Workflow
+
+TODO(incfly): just an illustrative note, will move to the README.md once implemented.
+
+High level, following tasks will be executed as periodic task from another Cloud Scheduler job via a separate webhook endpoint, `/flakeschaser`.
+
+- Issues matching following conditions (might needs adjustment based on fitering results)
+  - Regex matching in issue descrption or title, `.*test.*flakes` or with certain testing labels, `label/flake-tests`.
+  - Not closed yet
+  - `time.Now - last_updates_timestamp` > threath_hold, 3 days.
+- Update the issue with comments "@user-id testing flakes needs your attetion."
+- Send email to the assignee? Maybe later?
+
+## Code Changes
+
+- `plugins/flakeschaser` as pkg.
+- Read spanner table to figure out the working set.
+- Process working item by sending comments as Github client, Github throttler
+might be needed.
```

Comment:
If you follow the same pattern as in the rest of the code, you end up throttled naturally. The part that doesn't work is that I haven't implemented the 5000 QPH limit, and the limit is for a given access token so if there are multiple instances hitting GH at the same time, we might hit quota limits.

I don't have a strategy in place to deal with getting quota errors. It's probably worth having some standard way to do this systematically.

Reply:
sg. How about my other thought, 

> - Might better to update the Spanner tables right away to reduce the chance
another server instance duplicates `flakes chaser` comments. Alternatively, just invoke `/sync` by itself to simplfy the code logic?

without this, spanner `issuecomments` table can lag behind right after we post updates to neg people. If before that happened, we might neg people twice.

we can also not worry about this initially until it becomes a problem...

Reply:
I've got a PR pending that makes many improvements in this area.

'syncer' tries to sync everything as before. It should run once a day.

'refresher' listens to GitHub events as they occur and then updates Spanner. This means that the syncer should usually be close to a NOP.

In that world, we just need to make the refresher plugin run before any other GitHub plugins so ensure that Spanner is up to date enough. I think anyway.